lib/HLSL/DxilOutputColorBecomesConstant.cpp

```diff
@@ -0,0 +1,176 @@
+//                                                                           //
+// DxilOutputColorBecomesConstant.cpp                                        //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Provides a pass to stomp a pixel shader's output color to a given         //
+// constant value                                                            //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#include "dxc/HLSL/DxilGenerationPass.h"
+#include "dxc/HLSL/DxilOperations.h"
+#include "dxc/HLSL/DxilSignatureElement.h"
+#include "dxc/HLSL/DxilModule.h"
+#include "dxc/Support/Global.h"
+#include "dxc/HLSL/DxilTypeSystem.h"
+#include "dxc/HLSL/DxilInstructions.h"
+#include "dxc/HLSL/DxilSpanAllocator.h"
+
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/IntrinsicInst.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/ADT/BitVector.h"
+#include "llvm/Pass.h"
+#include "llvm/Transforms/Utils/Local.h"
+#include <memory>
+#include <unordered_set>
+
+using namespace llvm;
+using namespace hlsl;
+
+class DxilOutputColorBecomesConstant : public ModulePass {
+
+  enum VisualizerInstrumentationMode
+  {
+    PRESERVE_ORIGINAL_INSTRUCTIONS,
+    REMOVE_DISCARDS_AND_OPTIONALLY_OTHER_INSTRUCTIONS
+  };
+
+  float Red = 1.f;
+  float Green = 1.f;
+  float Blue = 1.f;
+  float Alpha = 1.f;
+  VisualizerInstrumentationMode Mode;
+
+  bool convertTarget0ToConstantValue(Function * OutputFunction, const hlsl::DxilSignature &OutputSignature, OP * HlslOP, float * color);
+
+public:
+  static char ID; // Pass identification, replacement for typeid
+  explicit DxilOutputColorBecomesConstant() : ModulePass(ID) {}
+  const char *getPassName() const override { return "DXIL Constant Color Mod"; }
+  void applyOptions(PassOptions O) override;
+  bool runOnModule(Module &M) override;
+};
+
+void DxilOutputColorBecomesConstant::applyOptions(PassOptions O)
+{
+  for (const auto & option : O)
+  {
+    if (0 == option.first.compare("constant-red"))
+    {
+      Red = atof(option.second.data());
+    }
+    else if (0 == option.first.compare("constant-green"))
+    {
+      Green = atof(option.second.data());
+    }
+    else if (0 == option.first.compare("constant-blue"))
+    {
+      Blue = atof(option.second.data());
+    }
+    else if (0 == option.first.compare("constant-alpha"))
+    {
+      Alpha = atof(option.second.data());
+    }
+    else if (0 == option.first.compare("mod-mode"))
+    {
+      Mode = static_cast<VisualizerInstrumentationMode>(atoi(option.second.data()));

```

Comment:
Yes. I wanted to make one change on the PIX side that passed these params in and be able to test it. My next change will be to implement the mod modes.